engine/src/lib.rs

```diff
@@ -317,6 +318,55 @@ mod contract {
         );
     }
 
+    /// Propose new contract bytecode for proxy contracts.
+    ///
+    /// Anyone can submit a proposal. The user submitting the proposal should cover the storage
+    /// staking fee. The proposal is used when the admin invokes `factory_accept` using
+    /// the hash (sha256) of this proposal as an argument.
+    #[no_mangle]
+    pub extern "C" fn factory_propose() {
```

Comment:
I have some questions about this method, namely around spam protection. If this function is open for anyone to call, then how would we deal with someone spamming us with random "proposals"? They would eat up our NEAR storage staking room and thus prevent real EVM transactions from going through.

I think the easiest way to avoid this problem would be to make this function owner-only as well, but then it seems pointless to have a propose/accept cycle because the owner would be performing both actions.

Another possibility would be to require calls to `factory_propose` to have attached enough NEAR to cover the storage for their proposal, but then we need to compute how much NEAR that actually is. We probably would also want there to be some logic to refund the NEAR after the proposal is accepted, and a way to remove your proposal (again to get your NEAR back). This adds complexity to the flow, and I am not sure if that complexity is worthwhile. For example, if we only allow owners to change the factory code (remove propose entirely), but the contract is controlled by a DAO then there could still be a propose/accept process; it would just be in the DAO contract instead. This would have the same functionality from a user's perspective, but I think splits up the complexity a little better because the DAO already needs this kind of propose/accept functionality for voting in the first place.

Reply:
Discussed with @mfornet and we agreed to remove `propose`+`accept` in favour of a single `update` method.